cl/corpus_importer/management/commands/harvard_opinions.py

```diff
@@ -566,57 +635,102 @@ def clean_docket_number(docket_number: str) -> str:
     :param docket_number: The docket number
     :param case_name_full: The full case name
     :return: OpinionCluster or None
     """
-    harvard_characters = clean_body_content(case_body)
     for case in possible_cases:
         cl_case_body = get_opinion_content(case)
         cl_characters = clean_body_content(cl_case_body)
+        if len(cl_characters) == 0:
+            logger.info(f"Empty Courtlistener opinion cluster: {case.id}")
+            continue
+
         diff = len(harvard_characters) / len(cl_characters)
         if not (0.3 < diff < 3):
             # Content too dissimilar in length to compare
             continue
 
         percent_match = compare_documents(harvard_characters, cl_characters)
-        if percent_match < 45:
+        if percent_match < 60:
             continue
-        if percent_match < 75 and len(harvard_characters) > 500:
-            # Require a name overlap for good but not great matches.
-            overlaps = overlap_case_names(case.case_name, case_name_full)
-            if not overlaps:
-                continue
-        elif percent_match < 98 and len(harvard_characters) < 500:
+
+        # Require some overlapping case title
+        overlaps = overlap_case_names(
+            case.case_name, [case_name_full, case_name_abbreviation]
+        )
+        if not overlaps:
+            continue
+
+        vector_match = get_cosine_similarity(
+            case.case_name, [case_name_full, case_name_abbreviation]
+        )
+        if vector_match < 0.3:
+            continue
+
+        # The threshold for washington state cases was around 500 but for florida
+        # it appears to be around 650.  Bumping the threshold for more intense
+        # comparisons to 1000 to be safe.
+        if len(harvard_characters) < 1000:
+            # Florida uses some pretty rote language in the ~650 character
+            # length that requires a bump in the length for stricter checking.
+
+            # This also means the matching threshold has to go for small cases
+            # completely so a 98% match in washington is not the
+            # same as a 99% match in Florida
+
             # Require a very close match - with name overlap and
             # docket number for very small cases.
             if percent_match < 90:
                 continue
-            overlaps = overlap_case_names(case.case_name, case_name_full)
-            if not overlaps:
-                continue
-            clean_docket = clean_docket_number(docket_number)
-            if clean_docket not in case.docket.docket_number:
+
+            # If a docket number exists: check against it.
+            if case.docket.docket_number is not None:
+                clean_docket = clean_docket_number(docket_number)
+                if clean_docket not in case.docket.docket_number:
+                    continue
+
+            # If you make it this far - we should check if this small case has
```

Comment:
You have great comments in this section, but I think a link here would help. 

I also wonder: Why not do this filter earlier and separately? I mentioned this in our chat today, but why not make this part of the finding of candidates on lines 778 and 798? You can do that exclusion earlier and have a smaller set to risk false positives on.